developer-experience/readme.md

```diff
@@ -0,0 +1,203 @@
+
+The following are ceremonies that the DevEx champion can use to find potential opportunities
+
+- Retrospectives. Is feedback being raised that relates to the essential tasks being difficult or unwieldy?
+- Standup Blockers. Are individuals getting blocked or stumbling on the essential tasks?
+
+As opportunities are identified, the DevEx champion can translate these into actionable stories for the product backlog.
+
+### Single IDE Instance
+
+Avoid requiring multiple instances of the IDE to running concurrently to run and debug the solution locally. This will negatively impact the above measure. Additionally, that negative impact is multiplied by each instance required.
+
+- Configuration steps * IDE instances
+- Build steps * IDE instances
+- Start/Debug steps * IDE instances
+- Stop steps * IDE instances
+- Run test steps * IDE instances
+- Documenting all of the above * IDE instances
+
+### Minimize the Number of Repositories
+
+Splitting a solution across multiple repositories can negatively impact the above measures. This can also negatively impact other areas such as Pull Requests, Automated Testing, Continuous Integration, and Continuous Delivery. Similar to the IDE instances, the negative impact is multiplied by the number of repositories.
+
+- clone steps * # of repos
+- branching steps * # of repos
+- commit steps * # of repos
+- CI * # of repos
+- Pull request reviews & merges * # of repos
+
+#### Atomic Pull Requests
+
+When the solution is encapsulated within a single repository, it also allows pull requests to represent a change across multiple layers. This is especially helpful when a change requires changes to a shared contract between multiple components. For example, a story requires that an api endpoint is changed. With this strategy the api and web client could be updated with the same pull request. This avoids master being broken temporarily while waiting on dependent pull requests to merge.
+
+[`TODO: Add link to other documented benefits of consolidated repos within source control section`]
+
+### When to Split Into Multiple Repositories
+
+[`TODO: Add link to source control section if it has guidance on this subject`]
+
+(Typically try to have the repository align with the product backlog.)
+
+### Minimize Remote Dependencies for Local Development
+
+The fewer dependencies on components that cannot run a developer's machine translate to fewer steps required to get started. Therefore, fewer dependencies will positively impact the measures above.
+
+The following strategies can be used to reduce these dependencies
+
+#### Use an Emulator
+
+If available, emulators are implementations of technologies that are typically only available in cloud environments. A good example is the [CosmosDB emulator](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator).
+
+#### Use DI + Toggle to Mock Remote Dependencies

```

Comment:
This is a pretty deep topic, is there any guidance we can link to instead of adding code samples here?

Reply:
I did a quick search to see if i could find any existing write-ups for this specific use case but was not able to. Since the code sample and supporting guidance is already written could we allow the first version to keep this? I think if we find ourselves adjusting this section in the future we should probably spend more effort finding a relevant article or, write our own that we can link to. Looks like there might be a sharing opportunity here since i wasnt able to initially find what i was looking for.